### DISCUSSION TITLE
Getting SSG circular reference errors starting in 9.3

### DISCUSSION THREAD

**Discussion by sdornan** [vercel/next.js#10905] [Help] [answered]
I'm currently using Next 9.2.3 with `unstable_getStaticProps` with Contentful without any issues. I attempted to upgrade to 9.3 now that `getStaticProps` has been officially released.

Doing so, however, gives me a `Reason: Circular references cannot be expressed in JSON.` without any code changes. My data does have some circular references in it, but I go out of my way using `safe-stringify` to remove them when the data is fetched from Contentful. 

What changed between 9.2.3 and 9.3? Is it possible to tell Next to ignore the result of the circular reference test if I'm sure the data is serializable, as it was in 9.2.3?

**Comment by ijjk** [comment, marked as answer]
Hi, this check was changed in `v9.3.1-canary.0` of Next.js. Can you check if you are you still seeing this error on that version?

**Reply by sdornan** [reply]
Works fine in `v9.3.1-canary.0`. Thanks!

**Comment by WEEFAA** [comment]
Hi, I'm on Next 9.5.0 with ```getServerSideProps``` with Contentful and it seems that I'm getting the same issues same as @sdornan does.  

I'm expecting to not have this error since I'm on the latest version of Next. How does ```v9.3.1-canary.0``` actually solve this?
